### PR TITLE
Add kms key table. Closes #124

### LINKED ISSUE #124: Add kms key table

### PR DESCRIPTION
**Pull request by bigdatasourav**
# Integration test logs
<details>
  <summary>Logs</summary>
  
```
SETUP: tests/gcp_kms_key []

PRETEST: tests/gcp_kms_key

TEST: tests/gcp_kms_key
Running terraform
google_kms_key_ring.named_test_resource: Creating...
google_kms_key_ring.named_test_resource: Creation complete after 4s [id=projects/pikachu-aaa/locations/global/keyRings/turbottest28373]
google_kms_crypto_key.named_test_resource: Creating...
google_kms_crypto_key.named_test_resource: Creation complete after 3s [id=projects/pikachu-aaa/locations/global/keyRings/turbottest28373/cryptoKeys/turbottest28373]

Warning: Deprecated Resource

The null_data_source was historically used to construct intermediate values to
re-use elsewhere in configuration, the same can now be achieved using locals


Apply complete! Resources: 2 added, 0 changed, 0 destroyed.

Outputs:

resource_aka = "gcp://cloudkms.googleapis.com/projects/pikachu-aaa/locations/global/keyRings/turbottest28373/cryptoKeys/turbottest28373"
resource_name = "turbottest28373"

Running SQL query: test-get-query.sql
[
  {
    "name": "turbottest28373",
    "rotation_period": "100000s"
  }
]
✔ PASSED

Running SQL query: test-list-query.sql
[
  {
    "name": "turbottest28373",
    "rotation_period": "100000s"
  }
]
✔ PASSED

Running SQL query: test-not-found-query.sql
null
✔ PASSED

Running SQL query: test-turbot-query.sql
[
  {
    "akas": [
      "gcp://cloudkms.googleapis.com/projects/pikachu-aaa/locations/global/keyRings/turbottest28373/cryptoKeys/turbottest28373"
    ],
    "title": "turbottest28373"
  }
]
✔ PASSED

POSTTEST: tests/gcp_kms_key

TEARDOWN: tests/gcp_kms_key

SUMMARY:

1/1 passed.
```
</details>

# Example query results
<details>
  <summary>Results</summary>
  
```
> select
  name,
  create_time,
  rotation_period
from
  gcp_kms_key;
+-------------------------+---------------------+-----------------+
| name                    | create_time         | rotation_period |
+-------------------------+---------------------+-----------------+
| turbottest33297         | 2021-03-22 09:36:09 |                 |
| test-key1               | 2021-02-05 10:35:32 |                 |
| test-key1               | 2021-02-05 10:33:08 | 7776000s        |
| turbottest2652          | 2021-01-22 09:00:59 |                 |
| turbottest29055         | 2021-05-25 11:13:03 |                 |
| turbottest19399         | 2021-05-25 11:06:03 |                 |
| turbottest25145         | 2021-05-25 11:37:28 |                 |
| test1                   | 2020-08-27 14:30:08 | 7776000s        |
| turbottest81132         | 2021-05-25 11:46:38 |                 |
| turbottest28373         | 2021-05-25 11:53:57 |                 |
| turbottest15188         | 2021-03-22 09:17:26 | 100000s         |
| turbottest56581         | 2021-03-22 07:47:15 | 100000s         |
| test-dataset-encryption | 2021-05-07 06:58:18 | 7776000s        |
| multiregion-dataset-key | 2021-05-07 07:01:20 |                 |
+-------------------------+---------------------+-----------------+
> select
  name,
  create_time,
  rotation_period
from
  gcp_kms_key
where
  create_time <= (current_date - interval '30' day)
order by
  create_time;
+-----------------+---------------------+-----------------+
| name            | create_time         | rotation_period |
+-----------------+---------------------+-----------------+
| test1           | 2020-08-27 14:30:08 | 7776000s        |
| turbottest2652  | 2021-01-22 09:00:59 |                 |
| test-key1       | 2021-02-05 10:33:08 | 7776000s        |
| test-key1       | 2021-02-05 10:35:32 |                 |
| turbottest56581 | 2021-03-22 07:47:15 | 100000s         |
| turbottest15188 | 2021-03-22 09:17:26 | 100000s         |
| turbottest33297 | 2021-03-22 09:36:09 |                 |
+-----------------+---------------------+-----------------+
> select
  name,
  create_time,
  rotation_period
from
  gcp_kms_key
where
  rotation_period = '100000s'
+-----------------+---------------------+-----------------+
| name            | create_time         | rotation_period |
+-----------------+---------------------+-----------------+
| turbottest56581 | 2021-03-22 07:47:15 | 100000s         |
| turbottest15188 | 2021-03-22 09:17:26 | 100000s         |
+-----------------+---------------------+-----------------+
```
</details>

### CODE REVIEW

File: docs/tables/gcp_kms_key.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+```sql
+select
+  name,
+  create_time,
+  rotation_period
+from
+  gcp_kms_key;
+```
+
+### List of keys older than 30 days
```
Comment (Subhajit97) [inline docs/tables/gcp_kms_key.md:18]:
```suggestion
### List keys older than 30 days
```

File: gcp/table_gcp_kms_key.go [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+	"github.com/turbot/steampipe-plugin-sdk/grpc/proto"
+	"github.com/turbot/steampipe-plugin-sdk/plugin"
+	"github.com/turbot/steampipe-plugin-sdk/plugin/transform"
+	"google.golang.org/api/cloudkms/v1"
+)
+
+func tableGcpKmsKey(ctx context.Context) *plugin.Table {
+	return &plugin.Table{
+		Name:        "gcp_kms_key",
+		Description: "GCP Kms Key",
```
Comment (Subhajit97) [inline gcp/table_gcp_kms_key.go:16]:
```suggestion
		Description: "GCP KMS Key",
```

File: gcp/table_gcp_kms_key.go [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+			},
+			{
+				Name:        "next_rotation_time",
+				Description: "At next_rotation_time, the Key Management Service will automatically: 1. Create a new version of this CryptoKey. 2.Mark the new version as primary.",
+				Type:        proto.ColumnType_TIMESTAMP,
+				Transform:   transform.FromField("NextRotationTime").Transform(transform.NullIfZeroValue),
+			},
+			{
+				Name:        "rotation_period",
+				Description: "RotationPeriod: next_rotation_time will be advanced by this period when the service automatically rotates a key.",
```
Comment (Subhajit97) [inline gcp/table_gcp_kms_key.go:57]: Can we update the column description?

File: gcp/table_gcp_kms_key.go [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+	}); err != nil {
+		return nil, err
+	}
+
+	return nil, nil
+}
+
+//// HYDRATE FUNCTIONS
+
+func getKeyDetail(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
```
Comment (Subhajit97) [inline gcp/table_gcp_kms_key.go:136]:
```suggestion
func getKeyDetail(ctx context.Context, d *plugin.QueryData, _ *plugin.HydrateData) (interface{}, error) {
```

File: gcp/table_gcp_kms_key.go [hunk 5]
```diff
@@ -0,0 +1,182 @@
...
+import (
+	"context"
+	"strings"
+
+	"github.com/turbot/steampipe-plugin-sdk/grpc/proto"
+	"github.com/turbot/steampipe-plugin-sdk/plugin"
+	"github.com/turbot/steampipe-plugin-sdk/plugin/transform"
+	"google.golang.org/api/cloudkms/v1"
+)
+
```
Comment (Subhajit97) [inline gcp/table_gcp_kms_key.go:12]:
```suggestion

//// TABLE DEFINITION

```

File: docs/tables/gcp_kms_key.md [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+```sql
+select
+  name,
+  create_time,
+  rotation_period
+from
+  gcp_kms_key
+where
+  rotation_period = '100000s'
+```
```
Comment (Subhajit97) [inline docs/tables/gcp_kms_key.md:44]: Can we update this example to check `List keys with rotation period >90 days`

Comment (Subhajit97) [reply docs/tables/gcp_kms_key.md:44]: Also, can we add another example to check whether the `key` is publicly accessible or not?

File: gcp/table_gcp_kms_key.go [hunk 7]
```diff
@@ -0,0 +1,184 @@
...
+	ringName := d.KeyColumnQuals["key_ring_name"].GetStringValue()
+
+	resp, err := service.Projects.Locations.KeyRings.CryptoKeys.Get("projects/" + project + "/locations/" + location + "/keyRings/" + ringName + "/cryptoKeys/" + name).Do()
+	if err != nil {
+		return nil, err
+	}
+
+	return resp, nil
+}
+
```
Comment (Subhajit97) [inline gcp/table_gcp_kms_key.go:165]: Can we add another API to get the `IAM policy`?

**Review by Subhajit97 — CHANGES_REQUESTED** [review]
@bigdatasourav Please see comments, thanks!

File: gcp/table_gcp_kms_key.go [hunk 8]
```diff
@@ -0,0 +1,225 @@
...
+				Type:        proto.ColumnType_TIMESTAMP,
+				Transform:   transform.FromField("NextRotationTime").Transform(transform.NullIfZeroValue),
+			},
+			{
+				Name:        "rotation_period",
+				Description: "Next rotation time will be advanced by this period when the service automatically rotates a key.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "policy",
```
Comment (Subhajit97) [inline gcp/table_gcp_kms_key.go:63]:
```suggestion
				Name:        "iam_policy",
```

File: gcp/table_gcp_kms_key.go [hunk 9]
```diff
@@ -0,0 +1,225 @@
...
+
+	var param string
+	if h.Item != nil {
+		param = h.Item.(*cloudkms.CryptoKey).Name
+	} else {
+		name := d.KeyColumnQuals["name"].GetStringValue()
+		location := d.KeyColumnQuals["location"].GetStringValue()
+		ringName := d.KeyColumnQuals["key_ring_name"].GetStringValue()
+		param = "projects/" + project + "/locations/" + location + "/keyRings/" + ringName + "/cryptoKeys/" + name
+	}
```
Comment (Subhajit97) [inline gcp/table_gcp_kms_key.go:197]: Why we need this for hydrate function?

**Review by Subhajit97 — APPROVED** [review]
LGTM

**Review by rajmohanty17 — APPROVED** [review]
LGTM

**Review by Subhajit97 — APPROVED** [review]

**Review by cbruno10 — APPROVED** [review]

**Pull request merged**